### DISCUSSION TITLE
Loss FR and CDR regions after assemble clones.

### DISCUSSION THREAD

**Discussion by Angieiscute** [milaboratory/mixcr#1253] [Q&A] [answered]
Hi, 

When using the command `mixcr exportClones -o -t --chains IGH --impute-germline-on-export full.alignment.clns full.tsv` to export the full sequence from PE300 data, I noticed that there are several clones where FR1 to CDR2 contain only lowercase letters.
<img width="1387" alt="image" src="https://github.com/milaboratory/mixcr/assets/130473448/3bb135db-a05d-4707-a262-a8633c0db80f">
After that, I attempted to use `mixcr exportAlignmentsPretty` to export from the VDJCA  file and observed that many reads are actually covered with all the regions. (there are some reads that have gaps in FR1~CDR2)
![image](https://github.com/milaboratory/mixcr/assets/130473448/13416716-4ce8-437b-a84c-3c2bf268553a)

I also use `mixcr exportClonesPretty` to export CLNS files and found that the FR1~CDR2 regions are missing (same clone). 
I believe it should display the uppercase letters because the regions are covered in the alignment step.
![image](https://github.com/milaboratory/mixcr/assets/130473448/f4e0c4d2-4772-4662-871e-5ed926cd5115)

What's the possible reason for the assembling step to cause these results?

My full commands are 
```
mixcr  align --rna --preset generic-bcr-amplicon --species hsa -OallowPartialAlignments=true -OvParameters.geneFeatureToAlign=VTranscript --rigid-left-alignment-boundary --floating-right-alignment-boundary C --report alignmentReport.log R1.fastq.gz R2.fastq.gz alignment.vdjca
mixcr  assemble -OseparateByC=true --write-alignments --report clonReport.log alignment.vdjca alignment.clna
mixcr  assembleContigs --report clonReport.log alignment.clna full.alignment.clns
mixcr  exportClones -o -t --chains IGH --impute-germline-on-export full.alignment.clns full.tsv
```

**Comment by mizraelson** [comment]
Hi,

1) To backtrack the alignments that constitute the particular clone, you should execute the following command:

```
mixcr exportAlignments --cloneId --descrsR1 --descrsR2 alignment.clna alignmetns.tsv
```

This command will extract the necessary fields from the `alignment.clna` file, including clone IDs and read descriptors (R1 and R2). With this information, you can then backtrack the exact reads that were used to assemble the specific clone. Only then we can be sure that we are looking at the proper alignments that actually relate to the clone of interest. 

2) Nevertheless if you are using an amplicon-based protocol, there is no need to utilize `assembleContigs`. Instead, the appropriate approach is to assemble clones by the feature of interest using the following parameter and then export clones directly:

```
mixcr assemble --assemble-clonotypes-by VDJRegion ...
```
3) In the case of amplicon-based data, I recommend using the `analyze` command, as it streamlines the process and avoids the need to run each step separately. Here's an example command:

```
mixcr analyze generic-bcr-amplicon \
--species hsa \
--rigid-left-alignment-boundary \
--floating-right-alignment-boundary C \
--assemble-clonotypes-by VDJRegion \
--split-clones-by C \
R1.fastq.gz R2.fastq.gz \
output
```

If your protocol is indeed amplicon-based (e.g., 5'RACE) i recommend running the command above and see if the issue is still there.

Best regards,
Mark

**Reply by Angieiscute** [reply]
Hi Mark, 
Thanks for your reply. 

I have used `mixcr exportAlignments --cloneId --descrsR1 --descrsR2 alignment.clna alignmetns.tsv` to export the reads and look for the clone I am talking about as an example. There are 58 reads total and only 19 reads have "region_not_covered" on both FR1 and CDR1. If in this case, the assembling step should use the ones that have covered FR1 and CDR1 to export use uppercase letters, right?


<img width="1618" alt="image" src="https://github.com/milaboratory/mixcr/assets/130473448/04fce57c-6810-43be-b759-36bd21f00f04">

**Reply by mizraelson** [reply, marked as answer]
By default, MiXCR discards alignments, if for some nucleotides sequencing quality is lower than 15. In the table you provided, I noticed that many alignments have a `minQual` of 0 for certain regions. To preserve these low-quality alignments, you can add the following parameter to the `mixcr analyze` command:
`-Massemble.cloneAssemblerParameters.badQualityThreshold=0`.
